homeassistant/components/solaredge/const.py

```diff
@@ -149,35 +149,50 @@
         json_key="Purchased",
         name="Imported Power",
         entity_registry_enabled_default=False,
-        icon="mdi:flash",
+        icon="mdi:home-import-outline",
+        state_class=STATE_CLASS_MEASUREMENT,
+        unit_of_measurement=ENERGY_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
     ),
     SolarEdgeSensorEntityDescription(
         key="production_power",
         json_key="Production",
         name="Production Power",
         entity_registry_enabled_default=False,
-        icon="mdi:flash",
+        icon="mdi:solar-power",
+        state_class=STATE_CLASS_MEASUREMENT,
+        unit_of_measurement=ENERGY_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
     ),
     SolarEdgeSensorEntityDescription(
         key="consumption_power",
         json_key="Consumption",
         name="Consumption Power",
         entity_registry_enabled_default=False,
         icon="mdi:flash",
+        state_class=STATE_CLASS_MEASUREMENT,
+        unit_of_measurement=ENERGY_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
     ),
     SolarEdgeSensorEntityDescription(
         key="selfconsumption_power",
         json_key="SelfConsumption",
         name="SelfConsumption Power",
         entity_registry_enabled_default=False,
-        icon="mdi:flash",
+        icon="mdi:auto-download",
+        state_class=STATE_CLASS_MEASUREMENT,
+        unit_of_measurement=ENERGY_WATT_HOUR,
+        device_class=DEVICE_CLASS_ENERGY,
     ),
     SolarEdgeSensorEntityDescription(
         key="feedin_power",
         json_key="FeedIn",
         name="Exported Power",
         entity_registry_enabled_default=False,
-        icon="mdi:flash",
+        icon="mdi:home-export-outline",
+        state_class=STATE_CLASS_MEASUREMENT,
+        unit_of_measurement=ENERGY_WATT_HOUR,
```

Comment:
Use `native_unit_of_measurement` instead of `unit_of_measurement` in `SolarEdgeSensorEntityDescription`
When a `device_class` is defined, `icon` should not be defined.

Reply:
May you right. I'm not a coder and I just followed how was write the code, in order to fix the issue is involving so many users. Feel free to commit any changes you want to make. I don't have any knowledge to go ahead, thanks.

Reply:
1) You rename `unit_of_measurement` to `native_unit_of_measurement` in all `SolarEdgeSensorEntityDescription` objects.
2) You remove `icon` from all `SolarEdgeSensorEntityDescription` objects where you added `device_class`

Reply:
1. I dont' renamed anything. native_unit_of_measurement was never used on const.py.
2. I've only change icon from production_power, selfconsumption_power, feedin_power, to make them more consistent instead of mdi:flash

Reply:
> I dont' renamed anything.

Those need changing @terminet85

Reply:
Sorry but as I said is beyond my knowledge

Reply:
If you can't make this change, I suggest to close this PR.

Reply:
As you want. I hope some other dev will fix it, however you left so many user without this function.
Your choice.

Reply:
There are litterly hundres of integrations that needs to adjust to these new features. Everything in time. But if you can't make changes, please don't try to give something half that isn't finished or usable.

You have just wasted time of 2 people that could have been working on actually doing something else, like fixing another case instead of reviewing this.

Nevertheless, thank you for willing to contribute.

Reply:
No problem, anyway is finished and usable. I'm just using it now in my home-assistant box.
There should be more elasticity... You want to change a class when is already used. I'm not a dev, I want just give my two cents. 
There should be more understanding...

Reply:
Please understand that we cannot merge this, but you are also not capable to implement this correctly. This is not helping anybody.

It might be working, but it is currently not correctly implemented. Are you suggesting we should ship incorrect code to our users?

Please, this understanding works both ways. I'm happy to make the adjustment, but we all have a lot of changes to make and things to improve. Patience is key and things like this is not helping, but distracting.

Reply:
So apologize if I wasted your time, next time I'll waiting and I keep the code to myself, so anybody he gets angry. 
Good work.

Reply:
I'm not angry. :)